src/DRICH_geo.cpp

```diff
@@ -546,29 +574,94 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
-          std::string modsecName = secName + "_" + std::to_string(imod);
-          DetElement  sensorDE(det, "sensor_de_" + modsecName, imodsec);
-          sensorDE.setPlacement(sensorPV);
-          if (!debugOptics || debugOpticsMode == 3) {
-            SkinSurface sensorSkin(desc, sensorDE, "sensor_optical_surface_" + modsecName, sensorSurf, sensorVol);
-            sensorSkin.isValid();
-          };
-
-          // increment sensor module number
-          imod++;
-
-        }; // end patch cuts
-      };   // end phiGen loop
-    };     // end thetaGen loop
+	    auto        imodsec    = encodeSensorID(sensorPV.volIDs());
+	    std::string modsecName = secName + "_" + std::to_string(imod);
+	    DetElement  sensorDE(det, "sensor_de_" + modsecName, imodsec);
+	    sensorDE.setPlacement(sensorPV);
+	    if (!debugOptics || debugOpticsMode == 3 || debugOpticsMode == 5) {
+	      SkinSurface sensorSkin(desc, sensorDE, "sensor_optical_surface_" + modsecName, sensorSurf, sensorVol);
+	      sensorSkin.isValid();
+	    };
+
+	    // increment sensor module number
+	    imod++;
+
+	  }; // end patch cuts
+	};   // end phiGen loop
+      };     // end thetaGen loop
+    }; // end sensor loop
+
+    // large sensor placement (for focal point testing):
+    if (debugOpticsMode == 4){
+      Box    sensorSolidScaled(sensorRescale * sensorSide / 2., sensorRescale * sensorSide / 2., sensorThickness / 2.);
+      Volume sensorVolScaled(detName + "_sensor_" + secName, sensorSolidScaled, sensorMat);
+      sensorVolScaled.setVisAttributes(sensorVis);
+      sensorVolScaled.setSensitiveDetector(sens);
+
+      double phiGen = 0;
+      auto sensorPlacement =
+        RotationZ(sectorRotation) *
+        Translation3D(sensorSphPos.x()+100, sensorSphPos.y(), sensorSphPos.z()-125) *
+        RotationX(phiGen) *
+        Translation3D(sensorSphRadius, 0., 0.)*
+        RotationZ(0);
+      auto sensorPV = gasvolVol.placeVolume(sensorVolScaled, sensorPlacement);
+      sensorPV.addPhysVolID("sector", isec).addPhysVolID("module", imod);
+
+      DetElement sensorDE(det, Form("sensor_de%d_%d", isec, imod), 0);
+      sensorDE.setPlacement(sensorPV);
+
+      SkinSurface sensorSkin(desc, sensorDE, Form("sensor_optical_surface%d", isec), sensorSurf, sensorVolScaled);
+      sensorSkin.isValid();
+    }; // end large sensor placement
+
+    // for debugOpticsMode 5: drawing calculated focal points
+    // of thrown beams of parallel photons
+    if( std::getenv("FPPLOT_FILE")!=NULL && debugOpticsMode == 5){
+      //std::cout << "drawing focal points " << std::endl;
+      Cone FPsolid(1.,0.25,0.5,1,1);
+      Volume FPvol(detName + "_FPpos_" + secName, FPsolid, aerogelMat);
+      FILE * fptxt = fopen(std::getenv("FPPLOT_FILE"),"r");
+      double fpx, fpy, fpz, dirx, diry, dirz;
+      int fpnum=0;
+
+      while(fscanf(fptxt,"%lf %lf %lf %lf %lf %lf", &fpx, &fpy, &fpz, &dirx, &diry, &dirz)!=EOF){
```

Comment:
Please avoid files with 'just a bunch of numbers'. Use a file format that provides meta-data as part of the file. This could be xml as already supported by dd4hep, or it could be a json file. That will allow you to use more robust parameter files with key-value pairs.

Reply:
How high of a priority is this? Just because it would require changes to the other scripts as well, and anything demonstrating usage of this feature would make it clear what script/simulation setting in `drich-dev` generates these points.

Reply:
If I say that it's not a high priority, it won't ever get done, right?

Reply:
How about the following minimal requirements?
- The file reader must be able to ignore comment (full) lines anywhere in the file starting with a comment character `#`, so a preceding line in any file can (should) contain a description of the fields.
- The file reading cannot use C style `FILE` access but must use `ifstream`.